packages/react-flex/src/components/Flex/Flex.tsx

```diff
@@ -1,5 +1,80 @@
 import * as React from 'react';
+import { FlexProps, FlexSlots, FlexSlotProps } from './Flex.types';
+import { compose, createClassResolver, mergeProps } from '@fluentui/react-compose';
+import * as classes from './Flex.scss';
+import { FlexItem } from '../FlexItem/FlexItem';
+import { FlexContext } from './FlexContext';
+import { useFlex } from './useFlex';
 
-export const Flex = () => {
-  return <div />;
+export const Flex = compose<'div', FlexProps, FlexProps, {}, {}>(
+  (props, ref, options) => {
+    const { children, disableShrink } = props;
+    const { state } = options;
+    const { slots, slotProps } = mergeProps<FlexProps, FlexProps, FlexSlots, FlexSlotProps>(state, options);
+    const gap: string = props.tokens && props.tokens.gap ? props.tokens.gap : '0';
+
+    const { flexChildren, generalChildren } = separateFlexChildren(children);
+
+    return (
+      <slots.root {...slotProps.root}>
+        <FlexContext.Provider value={{ disableShrink: !!disableShrink, gap: gap }}>{flexChildren}</FlexContext.Provider>
+        {generalChildren}
+      </slots.root>
+    );
+  },
+  {
+    classes: createClassResolver(classes),
+    displayName: 'Flex',
+    handledProps: [
+      'inline',
+      'column',
+      'wrap',
+      'horizontalAlign',
+      'verticalAlign',
+      'reverse',
+      'disableShrink',
+      'fluid',
+      'tokens',
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ] as any,
```

Comment:
Why do we need this `any` cast?

Reply:
I saw it on `react-button` but I will remove it, it doesn't seem to need it.